### PR TITLE
DM-18356: Modify default template to make filenames unique

### PR DESCRIPTION
**Pull request by hsinfang**
Alternatively we can change this only in the BPS layer ([gen3_pegasus](https://github.com/lsst-dm/gen3_pegasus) ) and not here in daf_butler.

### CODE REVIEW

**Review by timj — APPROVED** [review]
I'm happy with making the default template usable. It was only added as a proof of concept to start with. I think I'd prefer removing all those useless directories though and retaining the first two layers (so collection and run and datasetType.component) part.

**Comment by hsinfang** [comment]
Sounds good. I got rid of some layers of directories, leaving those I think useful so that the numbers of files in one directory aren't getting too large too soon.

File: config/datastores/posixDatastore.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ datastore:
     # valid_first and valid_last here are YYYYMMDD; we assume we'll switch to
     # MJD (DM-15890) before we need more than day resolution, since that's all
     # Gen2 has.
-    default: "{collection}/{datasetType}.{component:?}/{skymap:?}/{tract:?}/{patch:?}/{label:?}/{abstract_filter:?}/{instrument:?}/{physical_filter:?}/{calibration_label:?}/{visit:?}/{exposure:?}/{detector:?}"
+    default: "{collection}/{datasetType}.{component:?}/{tract:?}/{patch:?}/{label:?}/{abstract_filter:?}/{physical_filter:?}/{visit:?}/{instrument:?}{run}_{skymap:?}_{datasetType}_{component:?}_{tract:?}_{patch:?}_{label:?}_{abstract_filter:?}_{physical_filter:?}_{calibration_label:?}_{visit:?}_{exposure:?}_{detector:?}"
```
Comment (timj) [inline config/datastores/posixDatastore.yaml]: Do you want instrument and run to run into each other?

Comment (hsinfang) [reply config/datastores/posixDatastore.yaml]: I actually did it on purpose but not for any good/real reasons (run is always a number).  Happy to have underscore there too.

Comment (hsinfang) [reply config/datastores/posixDatastore.yaml]: Some datatypes don't have `instrument` set; we don't mind having some filenames starting with `_` right?

Comment (timj) [reply config/datastores/posixDatastore.yaml]: Good point.  Maybe leave the underscore out then.  Not entirely sure what the correct fix is. Possibly remove strings between optional and mandatory if there are no `[a-zA-Z]` characters...

Comment (hsinfang) [reply config/datastores/posixDatastore.yaml]: Another take...  Filenames are now led by `datasetType`; probably better aesthetically speaking although it's only a personal preference.

**Pull request merged**